test/ml_methods/Clustering/test_agglomerative.py

```diff
@@ -0,0 +1,69 @@
+import os
+import pickle
+import shutil
+from unittest import TestCase
+
+import numpy as np
+from scipy import sparse
+from sklearn.cluster import AgglomerativeClustering as SklearnAgglomerative
+
+from immuneML.caching.CacheType import CacheType
+from immuneML.data_model.encoded_data.EncodedData import EncodedData
+from immuneML.environment.Constants import Constants
+from immuneML.environment.EnvironmentSettings import EnvironmentSettings
+from immuneML.ml_methods.Clustering.Agglomerative import Agglomerative
+from immuneML.util.PathBuilder import PathBuilder
+
+
+class TestAgglomerative(TestCase):
+
+    def setUp(self) -> None:
+        os.environ[Constants.CACHE_TYPE] = CacheType.TEST.name
+
+    def test_fit(self):
+        x = np.array([[1, 0, 0], [0, 1, 1], [1, 1, 1], [0, 1, 1]])
+
+        method = Agglomerative()
+        method.fit(EncodedData(examples=sparse.csr_matrix(x)))
+
+        self.assertTrue(method.model.labels_ is not None)
+
+    def test_store(self):
+        x = np.array([[1, 0, 0], [0, 1, 1], [1, 1, 1], [0, 1, 1]])
+
+        method = Agglomerative()
+        method.fit(EncodedData(examples=sparse.csr_matrix(x)))
+
+        path = EnvironmentSettings.root_path / "test/tmp/method/"
```

Comment:
fine if paths are unique, but better to have descriptive names to make sure they are :)